### DISCUSSION TITLE
Analyzing calls in DeepSqueak

### DISCUSSION THREAD

**Discussion by ande4808** [DrCoffey/DeepSqueak#174] [Q&A]
Hi,
Below is an image of one of the rat pup files that I'm trying to analyze in DeepSqueak. Why are the calls "stacked" on top of each other? Could that be an echo picked up by the microphone I used for USV detections? Are the calls in the higher frequency range not true calls?

Thanks!

![DeepSqueak 7_25_2022 4_34_04 PM](https://user-images.githubusercontent.com/97688378/180879546-57f926d7-ea66-45b0-8160-f88e3bd4e3ac.png)

**Comment by tuskinekinase** [comment]
I don't know the context, but based on frequency contour, it looks like harmonics.

**Reply by DrCoffey** [reply]
Yup, these are harmonics. Although they are boxed, deepsqueak will only analyze the loudest part of the contour at any time point, so you see only 1 line to the left. I tend to ignore them, but I know some people are interested in analyzing them.

**Reply by katze-lucky** [reply]
I'm now including the harmonics part for my calls for my rat pup calls analysis, coz we think the harmonics are part of the call and can show the some characteristics of the calls. But if it is as you said that the deepsqueak only analyze the loudest part of the call, does it mean that the analysis parameters like principle frequency, frequency modulation and peak frequency would be the same no matter I include the harmonic part or not.

**Reply by DrCoffey** [reply]
Yes, the analysis parameters are based on the contour, and the contour can only occupy 1 frequency per timepoint. There is 1 statistic in the output file that relays the total box height, so you can get some information about the harmonic that way.

**Reply by VoiceScientist** [reply]
@katze-lucky I've always wondered what the mechanism is for producing those harmonics. Colleagues of mine have hypothesized that all of the USVs have them, but they are just too weak at the higher frequencies. I'm curious what you are hypothesizing the relevance of the harmonic portion is (physiological, behavioral, etc.?). The contour of the harmonic should theoretically be identical to the fundamental (just 2x the frequency). Are you thinking the intensity of the harmonic has some meaning?

**Reply by katze-lucky** [reply]
@DrCoffey Thank you for your response! Could I ask which statistic relays the total box height? Is it slope or sinuosity? BTW, could I ask if the the principal frequency means the avarage of the peak frequencies(the frequency at the loudest point) for all the timepoints of one call?

**Reply by katze-lucky** [reply]
@VoiceScientist I am new to the USV field and I also have a guess that if all USV have harmonics. Since from the rat pup USV files I have went through for my project, I found most of the calls have harmonics. Maybe this is an interesting question for Rat USVs. But for other rodents like mice, the harmonics is not so often seen as in the rats. Here is a literature you might be interested in - https://doi.org/10.1371/journal.pone.0017460. the authors found that there is an increased proportion of calls having harmonics when the mice pups getting older. Maybe the harmonics can increase the complexity of the USV of the pups which is good for the communication between mother and pups so that mother can better take care of the pups. For the mechanism of producing the harmonics, I'm not very clear about that. But there is a book you might be also interested in and can find some answer - <Handbook of Ultrasonic Vocalization> by STEFAN M. BRUDZYNSKI
